arb_os/arbowner.mini

```diff
@@ -75,6 +101,144 @@ impure func arbowner_giveOwnership(topFrame: EvmCallFrame, calldata: ByteArray)
+    if (bytearray_size(calldata) != 4) {
+        evmOp_revert_knownCodePc(address(const::Address_ArbOwner), 100, 0, 0);
+    }
+
+    if let Some(recipient) = gasAccounting_getFeeRecipient() {
+        let mem = bytearray_set256(bytearray_new(0), 0, uint(recipient));
+        if (evmCallStack_setTopFrameMemory(mem)) {
+            evmOp_return(0, 32);
+        } else {
+            evmOp_revert_knownCodePc(address(const::Address_ArbOwner), 101, 0, 0);
+        }
+    } else {
+        evmOp_revert_knownCodePc(address(const::Address_ArbOwner), 102, 0, 0);
+    }
+}
+
+impure func arbowner_setFeeRecipient(topFrame: EvmCallFrame, calldata: ByteArray) { // (address)
+    if (bytearray_size(calldata) != 36) {
+        evmOp_revert_knownCodePc(address(const::Address_ArbOwner), 110, 0, 0);
+    }
+    let recipient = address(bytearray_get256(calldata, 4));
+    if let Some(_) = gasAccounting_setFeeRecipient(recipient) {
+        evmOp_return(0,0);
+    } else {
+        evmOp_revert_knownCodePc(address(const::Address_ArbOwner), 111, 0, 0);
+    }
+}
+
+impure func arbowner_getFeeRates(topFrame: EvmCallFrame, calldata: ByteArray) { // () -> (uint, uint, uint, uint)
+    if (bytearray_size(calldata) != 4) {
+        evmOp_revert_knownCodePc(address(const::Address_ArbOwner), 60, 0, 0);
+    }
+
+    if let Some(res) = gasAccounting_getFeeRates() {
+        let (num1, denom1, num2, denom2) = res;
+        let mem = bytearray_set256(
+            bytearray_set256(
+                bytearray_set256(
+                    bytearray_set256(
+                        bytearray_new(0),
+                        0,
+                        num1,
+                    ),
+                    32,
+                    denom1,
+                ),
+                64,
+                num2,
+            ),
+            96,
+            denom2,
+        );
+        if (evmCallStack_setTopFrameMemory(mem)) {
+            evmOp_return(0, 4*32);
+        } else {
+            evmOp_revert_knownCodePc(address(const::Address_ArbOwner), 61, 0, 0);
+        }
+    } else {
+        evmOp_revert_knownCodePc(address(const::Address_ArbOwner), 62, 0, 0);
+    }
+}
+
+impure func arbowner_setFeeRates(topFrame: EvmCallFrame, calldata: ByteArray) { // (uint, uint, uint, uint)
+    if (bytearray_size(calldata) != 4+4*32) {
+        evmOp_revert_knownCodePc(address(const::Address_ArbOwner), 70, 0, 0);
+    }
+    let num1 = bytearray_get256(calldata, 4);
+    let denom1 = bytearray_get256(calldata, 4+32);
+    let num2 = bytearray_get256(calldata, 4+2*32);
+    let denom2 = bytearray_get256(calldata, 4+3*32);
+    if let Some(_) = gasAccounting_setFeeRates(num1, denom1, num2, denom2) {
+        evmOp_return(0,0);
+    } else {
+        evmOp_revert_knownCodePc(address(const::Address_ArbOwner), 71, 0, 0);
+    }
+}
+
+impure func arbowner_getFeeMaxes(topFrame: EvmCallFrame, calldata: ByteArray) { // () -> (uint, uint, uint, uint)
```

Comment:
We should consider making this publicly callable

Reply:
Same as above.